File: couch-server/src/main/java/com/fronde/server/services/snarkactivity/SnarkActivityCriteria.java
package com.fronde.server.services.snarkactivity;

import static org.springframework.format.annotation.DateTimeFormat.ISO;

import com.fronde.server.domain.criteria.AbstractCriteria;
import java.util.Date;
import org.springframework.format.annotation.DateTimeFormat;

public class SnarkActivityCriteria extends AbstractCriteria {

  private String island;
  private String locationID;
  private String birdID;
  // assume dates in all criteria objects are dates and not date times
  @DateTimeFormat(iso = ISO.DATE)
  private Date fromDate;
  // assume dates in all criteria objects are dates and not date times
  @DateTimeFormat(iso = ISO.DATE)
  private Date toDate;
  private Boolean includeTrackAndBowl;
  private Boolean includeHopper;
  private Boolean includeNest;
  private Boolean includeRoost;

  public String getIsland() {
    return island;
  }

  public void setIsland(String island) {
    this.island = island;
  }

  public String getLocationID() {
    return locationID;
  }

  public void setLocationID(String locationID) {
    this.locationID = locationID;
  }

  public String getBirdID() {
    return birdID;
  }

  public void setBirdID(String birdID) {
    this.birdID = birdID;
  }

  public Date getFromDate() {
    return fromDate;
  }

  public void setFromDate(Date fromDate) {
    this.fromDate = fromDate;
  }

  public Date getToDate() {
    return toDate;
  }

  public void setToDate(Date toDate) {
    this.toDate = toDate;
  }

  public Boolean getIncludeTrackAndBowl() {
    return includeTrackAndBowl;
  }

  public void setIncludeTrackAndBowl(Boolean includeTrackAndBowl) {
    this.includeTrackAndBowl = includeTrackAndBowl;
  }

  public Boolean getIncludeHopper() {
    return includeHopper;
  }

  public void setIncludeHopper(Boolean includeHopper) {
    this.includeHopper = includeHopper;
  }

  public Boolean getIncludeNest() {
    return includeNest;
  }

  public void setIncludeNest(Boolean includeNest) {
    this.includeNest = includeNest;
  }

  public Boolean getIncludeRoost() {
    return includeRoost;
  }

  public void setIncludeRoost(Boolean includeRoost) {
    this.includeRoost = includeRoost;
  }

}

File: angular-client/src/app/snarkactivity/snarkactivity.criteria.ts
import { AbstractCriteria } from '../domain/criteria/abstract-criteria';
import { Params } from '@angular/router';

export class SnarkActivityCriteria extends AbstractCriteria {
  island: string;
  locationID: string;
  birdID: string;
  fromDate: Date;
  toDate: Date;
  includeTrackAndBowl: boolean;
  includeHopper: boolean;
  includeNest: boolean;
  includeRoost: boolean;

  populateFromParams(params: Params) {
    this.doPopulateFromParams(
      params,
      ['includeTrackAndBowl', 'includeHopper', 'includeNest', 'includeRoost'],
      ['pageSize', 'pageNumber'],
      ['fromDate', 'toDate'],
      []
    );
  }

  toParams() {
    return this.doToParams(this);
  }
}
